pulsar-client/src/main/java/com/yahoo/pulsar/client/impl/ConsumerImpl.java

```diff
@@ -503,6 +503,14 @@ void connectionOpened(final ClientCnx cnx) {
                 cnx.channel().close();
                 return null;
             }
+            
+            if (client.getConfiguration().getConnectionsPerBroker() == 0) {
```

Comment:
Wouldn't it be better to close the connection after it gets not used anymore instead than after errors?